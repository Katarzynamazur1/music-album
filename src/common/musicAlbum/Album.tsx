import { ADD_ITEM_TO_CART } from "../../state/shoppingCart/shoppingCartActions";
import { useAppDispatch, useAppSelector } from "../../state/stateHooks";
import AddToCartButton from "../AddToCartButton";
import ShoppingCartIcon from "../ShoppingCartIcon";
import AlbumDescription, { AlbumDescriptionProps } from "./AlbumDescription";
import AlbumImage from "./AlbumImage";
import AlbumNumber from "./AlbumNumber";

const albumStyle = {
  display: "flex",
  justifyContent: "space-between",
  alignItems: "center",
  margin: "30px 20px 30px 20px",
  fontSize: "12px",
  background: "#353a45",
  padding: "10px",
};

export interface AlbumProps {
  id: string;
  number: number;
  coverImageUrl: string | undefined;
  description: AlbumDescriptionProps;
  discountValue?: number;
}

const Album = ({
  id,
  number,
  coverImageUrl,
  description,
  discountValue,
}: AlbumProps) => {
  const dispatch = useAppDispatch();

  const quantityOfThisAlbumInShoppingCart = useAppSelector(
    (state) => state.shoppingCart.items.find((i) => i.id === id)?.quantity ?? 0
  );

  const addAlbumToCart = () => {
    const cartItem = {
      name: description.title,
      price: description.price,
      id: id, // mozna by tez zapisac syntaxem id (warto to wiedziec)
    };
    dispatch({ type: ADD_ITEM_TO_CART, item: cartItem });
  };

  return (
    <div data-testid="album" style={albumStyle}>
      <AlbumNumber value={number} />
      <AlbumImage imageUrl={coverImageUrl} salesStickerValue={discountValue} />
      <AlbumDescription
        title={description.title}
        artist={description.artist}
        price={description.price}
        percentDiscount={discountValue}
        releaseDate={description.releaseDate}
      />
      <ShoppingCartIcon
        numberOfItems={quantityOfThisAlbumInShoppingCart}
        transparent
      />
      <AddToCartButton onClick={addAlbumToCart} />
    </div>
  );
};

export default Album;
