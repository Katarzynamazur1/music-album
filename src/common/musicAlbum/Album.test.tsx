import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import store from "../../state/store";
import Album from "./Album";

describe("Album.tsx", () => {
  it("matches snapshot", () => {
    const testAlbumDescription = {
      title: "Placz Kotku",
      artist: "Behemot",
      price: 666,
    };

    const { container } = render(
      <Provider store={store}>
        <Album
          id={"konkursowe123"}
          number={666}
          coverImageUrl={"https://m.media-amazon.com/images/I/614ub5HQC2L.jpg"}
          description={testAlbumDescription}
        />
      </Provider>,
      { wrapper: BrowserRouter }
    );
    expect(container).toMatchSnapshot();
  });

  // Zadanie: jak mozemy przetestowac interakcje, kiedy klikniemy na "add" button?
});
