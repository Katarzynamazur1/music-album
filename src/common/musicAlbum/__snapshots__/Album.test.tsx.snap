// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Album.tsx matches snapshot 1`] = `
<div>
  <div
    data-testid="album"
    style="display: flex; justify-content: space-between; align-items: center; margin: 30px 20px 30px 20px; font-size: 12px; background: rgb(53, 58, 69); padding: 10px;"
  >
    <div
      style="display: flex; align-items: center; padding: 5px; margin: 5px; font-size: 30px; font-family: 'Brush Script MT', cursive; color: rgb(237, 92, 145);"
    >
      666
    </div>
    <div
      style="position: relative;"
    >
      <div
        style="width: 100px; min-width: 100px; height: 100px; background: blue; background-image: url(https://m.media-amazon.com/images/I/614ub5HQC2L.jpg); background-size: contain; border-radius: 50%; margin: 10px;"
      />
      <div
        style="position: absolute; left: 0px; top: 0px;"
      />
    </div>
    <div
      style="margin: 8px;"
    >
      <div
        style="color: rgba(216, 237, 92, 0.612); font-weight: bold; font-size: 18px; font-family: 'Courier New', monospace; width: 330px; text-overflow: ellipsis; overflow: hidden; white-space: nowrap;"
      >
        Placz Kotku
      </div>
      <div
        style="margin: 10px; font-weight: 200; font-size: 14px; width: 330px;"
      >
        Behemot
      </div>
      <span
        style="margin: 8px;"
      >
        <span
          style="padding: 0px 5px; background-color: inherit;"
        >
          666
        </span>
        <span
          data-testid="discounted-price"
          style="padding: 0px 5px; background-color: inherit;"
        />
      </span>
    </div>
    <div
      style="position: relative; padding: 5px;"
    >
      <a
        href="/shopping-cart"
        style="text-decoration: none; color: white; opacity: 0.5; font-size: 36px;"
      >
        ðŸ›’
      </a>
    </div>
    <button
      style="background: rgb(237, 92, 145); cursor: pointer; border-radius: 4px; padding: 12px;"
    >
      Add
    </button>
  </div>
</div>
`;
